erpnext/stock/doctype/stock_entry/stock_entry.py

```diff
@@ -213,6 +214,19 @@ def check_duplicate_entry_for_production_order(self):
 				frappe.throw(_("Stock Entries already created for Production Order ")
 					+ self.production_order + ":" + ", ".join(other_ste), DuplicateEntryForProductionOrderError)
 
+	def set_incoming_rate(self):
+		for d in self.items:
+			if d.s_warehouse:
+				args = self.get_args_for_incoming_rate(d)
+				d.basic_rate = get_incoming_rate(args)
+			elif d.t_warehouse and not d.basic_rate:
+				warehouse = d.s_warehouse or d.t_warehouse
```

Comment:
why d.s_warehouse or d.t_warehouse ? If there is s_warehouse this `elif` will not be executed